### PR TITLE
Kubernetes deployment

### PR DESCRIPTION
**Pull request by jorge0136**
Much of ongoing work that Hal has been working on was swept up into this PR. The majority of the work I am focusing on is in regards to deploying e3os-api to kubernetes. 

I am very happy with the way this has ended up working. We are able to patch the deployment in a simple way with `kustomize` and then use `skaffold` to deploy it to the cluster. 

As I am talking here, I am realizing a `readme` update is in order. Short version, to be expounded upon is: 

* Select which kubernetes cluster to push against. This is set via `kubecontext`. `skaffold` will behave differently depending on what kubernets cluster is connected. 

* `skaffold run` will put up the image with a managed `git SHA tag` and deploy the kubernetes manifests.

### CODE REVIEW

File: sample_code_e3os_api.py [hunk 1]
```diff
@@ -6,10 +6,10 @@ class APIHelper():
...
     def __init__( self ):
 
-        #self.SERVICE_ENDPOINT = 'http://localhost:5555'
-        self.SERVICE_ENDPOINT = 'https://e3osapi.optimumenergyco.com'
-        self.USER = 'cdemoll@prescriptivedata.io'
-        self.PASSWORD = 'PleaseChangeMe!'
+        #self.SERVICE_ENDPOINT = 'http://127.0.0.1:5555'
+        self.SERVICE_ENDPOINT = 'http://0.0.0.0:5555'
+        self.USER = 'hal-api-test'
+        self.PASSWORD = 'DWcm6AWm@Qrxw9L'
```
Comment (jorge0136) [inline sample_code_e3os_api.py]: Should this be in here Hal?

File: e3os-api.py [hunk 2]
```diff
@@ -75,128 +54,57 @@ def requires_auth(f):
...
-    if currentUser in allowedUsers:
-        return True
-    else:
-        return False
 
 @app.route("/oauth/token", methods=['POST'])
 def getToken():
 
     un = request.form.get("username")
     pw = request.form.get("password")
```
Comment (NolanAinsworth) [inline e3os-api.py]: I think these variable names could be a little clearer.

File: model/e3os_getDatapoints.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        finally:
+            if conn is not None:
+                conn.close()
+
+        return data
+
+    def myconverter(o):
+        if isinstance(o, datetime.datetime):
+            return o.__str__()
```
Comment (NolanAinsworth) [inline model/e3os_getDatapoints.py]: We should probably appease the linter and put a newline at the end of file here.

File: model/e3os_history_query.py [hunk 4]
```diff
@@ -11,13 +12,56 @@ def __init__(self, host, user, password):
...
+            , [FriendlyName]
+            , [DataPointName]
+    
+            FROM[oemvm].[dim].[DataPoint_List]
+            where SiteID = %s 
+            and isRollupPoint=0
+            and DataPointName in (%s)
+            ''' % (siteID, pointNamesString)
+
+        # print(queryString)
```
Comment (NolanAinsworth) [inline model/e3os_history_query.py]: Can this be removed?

File: sample_code_e3os_api.py (hunk 1 above)
Comment (optimumhw) [reply sample_code_e3os_api.py]: i will fix this

File: e3os-api.py (hunk 2 above)
Comment (optimumhw) [reply e3os-api.py]: ok, i'll fix that Nolan

File: model/e3os_getDatapoints.py (hunk 3 above)
Comment (optimumhw) [reply model/e3os_getDatapoints.py]: ok

File: model/e3os_history_query.py (hunk 4 above)
Comment (optimumhw) [reply model/e3os_history_query.py]: yes

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  e3os-api:
+    build: .
+    stdin_open: true
+    tty: true
+    ports:
+      - "5555:5555"
+    environment:
+      E3OS_USER: POPULATE_ME
+      E3OS_PASSWORD: POPULATE_ME
+      # TODO: Add `stage2sql` connection here.
```
Comment (jorge0136) [inline docker-compose.yml]: I think we can probably remove this TODO, right @optimumhw

File: e3os-api.py [hunk 6]
```diff
@@ -75,128 +54,57 @@ def requires_auth(f):
...
-                                    "Unable to parse authentication"
-                                    " token."}, 401)
-
-            _request_ctx_stack.top.current_user = payload
-            return f(*args, **kwargs)
-        raise AuthError({"code": "invalid_header",
-                        "description": "Unable to find appropriate key"}, 401)
-    return decorated
 
+        #print("token : " + token)
```
Comment (jorge0136) [inline e3os-api.py]: Do we need this commented code?

**Pull request merged**